lite-xl/lite-xl
# Double click EmptyView to make a new document (and other minor quality of life changes)

Feature changes:
- Can double click the empty view to open a new document
- Can double click a tab gutter (in the space where there isn't already a tab) to make a new document
- Can scroll the mouse wheel over a tab gutter to scroll tabs if more than 10 tabs are open

Technical changes:
- Moved EmptyView into a separate script. If a plugin ever wants to modify the EmptyView, this makes it possible.
- Light refactoring in rootview to break out some utility functions
- Added some helper functions to find out what part of the tab gutter a click was in

Risks:
- If any of my math is wrong, double clicks outside of the tab gutter could open new documents
- If the scrolling logic has a bug, it could eat scroll events that are meant to go deeper into the view hierarchy

Testing:
1. Build+Run lite xl
2. Close all tabs
3. Double click the background, confirm a tab is opened
4. Double click the tab gutter, confirm a tab is opened (you may have to edit the existing document first - if there's already a new document, lite xl doesn't want to make another new one)
5. Double click a tab and confirm a new tab isn't opened. Open 10+ tabs and double click the arrows left/right of the tabs, confirm a new tab isn't opened.
6. Create an h-split + v-split. Confirm the tab gutters in a vertical/horizontal split can be double clicked as expected.
7. Open 10+ tabs in all splits. Try scrolling over the tab bar. Confirm the tab bar scrolls.
8. Open a long document in each view. Confirm you can still scroll.
9. Try double clicking around other views (especially the tree view) and confirm you can't accidentally open more new docs.
